### DISCUSSION TITLE
Looking to use pagy with Algolia, starting the page count at 0

### DISCUSSION THREAD

**Discussion by misham** [ddnexus/pagy#744] [Q&A] [answered]
Hi,

Trying to figure out how to use pagy with Algolia Ruby client. Not particularly hard, here's the approach:

```ruby
      @records = MyModel.search(params[:query])
      count = @records.raw_answer['nbHits']
      page = @records.raw_answer['page']
      limit = @records.raw_answer['hitsPerPage']
      @pagy = Pagy.new(count: count, page: page, limit: limit)
```

Except Algolia provides page number as zero-based. This causes pagy to throw an `Pagy::VariableError` with message of `expected :page >= 1; got 0 `.

I can hack it to just add 1 or subtract 1 in the code above, but wondering if there's a way to allow pagy to use zero-based count.

**Comment by ddnexus** [comment, marked as answer]
AFAIK the pagination business is conventionally all about pages, not page-offset, so no, there's no native support for that.  

It looks like you mean that the API provides a page-offset, not a page number.... interestingly there are a few widgets for pagination that actually show the one-based page number in the UI. (https://www.algolia.com/developers/code-exchange/pagination-widget/)... 

Maybe that's the official way of doing it? Maybe a pagy extra would be simple to write.

**Reply by misham** [reply]
Thanks, I figured that's the case, but wanted to ask. Good point on page offset, wasn't thinking of it that way.

Once I get a little more sleep, I'll look at adding Algolia support, a la ElasticSearch. The code looked straightforward enough.

**Reply by ddnexus** [reply]
FYI: there's also a [pagy](https://github.com/algolia/algoliasearch-rails/blob/master/lib/algoliasearch/pagination/pagy.rb) module that seems to support some minimal pagination, but it's not up-to-date.
